react/react-tests.ts

```diff
@@ -240,18 +247,15 @@ domNode = ReactDOM.findDOMNode(domNode);
 
 var type: React.ComponentClass<Props> = element.type;
 var elementProps: Props = element.props;
-var key: React.Key = element.key;
-
-var t: React.ReactType;
-var name = typeof t === "string" ? t : t.displayName;
+var key = element.key;
 
 //
 // React Components
 // --------------------------------------------------------------------------
 
-var displayName: string = ClassicComponent.displayName;
-var defaultProps: Props = ClassicComponent.getDefaultProps();
-var propTypes: React.ValidationMap<Props> = ClassicComponent.propTypes;
+var displayName: string | undefined = ClassicComponent.displayName;
+var defaultProps: Props | {} = ClassicComponent.getDefaultProps ? ClassicComponent.getDefaultProps() : {};
```

Comment:
can you keep `: Props =` and cast the empty object (`<Props>{}`) instead? Otherwise, if `getDefaultProps()` has the wrong return type it'll get assigned to `{}` and won't break.